ratis-grpc/src/main/java/org/apache/ratis/grpc/server/GrpcService.java

```diff
@@ -118,23 +144,76 @@ private GrpcService(RaftServer raftServer, Supplier<RaftPeerId> idSupplier, int
 
     this.clientProtocolService = new GrpcClientProtocolService(idSupplier, raftServer);
 
+    final int port = serverConfig.getPort();
     this.serverInterceptor = new MetricServerInterceptor(
         idSupplier,
         JavaUtils.getClassSimpleName(getClass()) + "_" + port
     );
 
-    NettyServerBuilder nettyServerBuilder = NettyServerBuilder.forPort(port)
+    final boolean separateAdminServer = !serverConfig.equals(adminServerConfig);
+    final boolean separateClientServer = !serverConfig.equals(clientServerConfig);
+
+    final NettyServerBuilder serverBuilder =
+        startBuildingNettyServer(serverConfig, grpcMessageSizeMax, flowControlWindow);
+    serverBuilder.addService(ServerInterceptors.intercept(
+        new GrpcServerProtocolService(idSupplier, raftServer), serverInterceptor));
+    if (!separateAdminServer) {
+      addAdminService(raftServer, serverBuilder);
```

Comment:
Let's move this line down to the else-clause (line 179) below?

Reply:
I don't think we can move it after `serverBuilder.build()` call.

Reply:
Good point.  Please ignore my previous comment and keep it this way.